xbmc/addons/kodi-addon-dev-kit/include/kodi/addon-instance/ShaderPreset.h

```diff
@@ -1,148 +1,116 @@
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation; either version 2, or (at your option)
+ *  any later version.
+ *
+ *  This Program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this Program; see the file COPYING.  If not, see
+ *  <http://www.gnu.org/licenses/>.
+ *
+ */
 #pragma once
-/*  Copyright (C) 2017 Team Kodi
-*
-*  RetroArch - A frontend for libretro.
-*  Copyright (C) 2010-2014 - Hans-Kristian Arntzen
-*  Copyright (C) 2011-2017 - Daniel De Matteis
-*
-*  RetroArch is free software: you can redistribute it and/or modify it under the terms
-*  of the GNU General Public License as published by the Free Software Found-
-*  ation, either version 3 of the License, or (at your option) any later version.
-*
-*  RetroArch is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
-*  without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR
-*  PURPOSE.  See the GNU General Public License for more details.
-*
-*  You should have received a copy of the GNU General Public License along with RetroArch.
-*  If not, see <http://www.gnu.org/licenses/>.
-*/
 
 #include "../AddonBase.h"
 
-#include <stdbool.h>
+#include <stdint.h>
 
 namespace kodi { namespace addon { class CInstanceShaderPreset; } }
 
-extern "C"
-{
-  /// ======== config_file_ ========
-
-  struct config_entry_list_
-  {
-    /* If we got this from an #include,
-    * do not allow overwrite. */
-    bool readonly;
-    char *key;
-    char *value;
-    uint32_t key_hash;
-
-    struct config_entry_list_ *next;
-  };
-
-  struct config_include_list_
-  {
-    char *path;
-    struct config_include_list_ *next;
-  };
-
-  struct config_file_
-  {
-    char *path;
-    struct config_entry_list_ *entries;
-    struct config_entry_list_ *tail;
-    unsigned include_depth;
-
-    struct config_include_list_ *includes;
-  };
-
-  typedef struct config_file_ config_file_t_;
-
-  /// =============================
-
-
-  /// ==== video_shader_PARSE =====
-
 #ifndef PATH_MAX_LENGTH
-#if defined(_XBOX1) || defined(_3DS) || defined(PSP) || defined(GEKKO)|| defined(WIIU)
-#define PATH_MAX_LENGTH 512
-#else
 #define PATH_MAX_LENGTH 4096
 #endif
-#endif
 
-#ifndef GFX_MAX_SHADERS
 #define GFX_MAX_SHADERS 26
-#endif
-
-#ifndef GFX_MAX_TEXTURES
 #define GFX_MAX_TEXTURES 8
-#endif
-
-#ifndef GFX_MAX_VARIABLES
 #define GFX_MAX_VARIABLES 64
-#endif
-
-#ifndef GFX_MAX_PARAMETERS
 #define GFX_MAX_PARAMETERS 128
-#endif
 
-  struct state_tracker_uniform_info_
+extern "C"
+{
+  typedef struct config_file config_file;
+
+  typedef enum state_tracker_type
+  {
+    RARCH_STATE_CAPTURE = 0,
```

Comment:
The point of the `_` was to solve conflicts with the normal enum definitions.
I was getting conflicts without them.
Have you tried to compile like that and if so, how did you solve the issue?